Cargo.toml

```diff
@@ -28,9 +28,11 @@ openssl-sys = { version = "0.9.68", optional = true }
 [features]
 default = ['tor']
 # Mac OS needs to use the vendored version of openssl-sys as upstream `libtor` is not updated for to support mac os aarch (M series)
-build-macos-aarch = ["openssl-sys/vendored"]
+build-macos-aarch = ['openssl-sys/vendored', 'dep:libtor', 'dep:mitosis']
 # The following feature set is in response to the issue described at https://github.com/rust-lang/rust/issues/45599
 # Only used for running the integration tests
 integration-test = []
+# Mac OS needs to use the vendored version of openssl-sys for integrating testing as upstream `libtor` is not updated for to support mac os aarch (M series)
+integration-test-macos-aarch = ['openssl-sys/vendored']
```

Comment:
This also needs to be mentioned in the readme, as a note for the test command.

Reply:
Do we need another feature for this? 

We should be able to do just `cargo test --features=integration-test, build-macos-arch -- --no-cpature`?

Reply:
- Will update the docs
- testing with `--features=integration-test` doesnt include `tor` libs but do include the `openssl` which is causing issues for `integration-test` on macos, hence we need to have `macos specific integration testing` due to `openssl upstream`. Macos only with `aarch`